src/main/java/org/folio/bulkops/service/ErrorService.java

```diff
@@ -138,35 +138,27 @@ public Errors getErrorsPreviewByBulkOperationId(UUID bulkOperationId, int limit,
     }
   }
 
-  public void saveErrorsFromDataImport(UUID bulkOperationId, UUID dataImportJobId) {
-    log.info("Saving errors from DataImport, bulkOperationId = {}, dataImportJobId = {}", bulkOperationId, dataImportJobId);
-    var bulkOperation = operationRepository.findById(bulkOperationId)
-      .orElseThrow(() -> new NotFoundException("BulkOperation was not found by id=" + bulkOperationId));
-    var identifierType = bulkOperation.getIdentifierType();
-    try {
-      var jobLogEntries = metadataProviderClient.getJobLogEntries(dataImportJobId.toString(), Integer.MAX_VALUE)
-        .getEntries().stream()
-        .filter(entry -> nonNull(entry.getError()))
-        .toList();
-      jobLogEntries.forEach(errorEntry -> {
-        List<String> identifierList = null;
-        var relatedInstanceInfo = errorEntry.getRelatedInstanceInfo();
-        if (identifierType == IdentifierType.ID) {
-          identifierList = relatedInstanceInfo.getIdList();
-        } else if (identifierType == IdentifierType.INSTANCE_HRID) {
-          identifierList = relatedInstanceInfo.getHridList();
-        }
-        var identifier = CollectionUtils.isEmpty(identifierList) ? null : identifierList.get(0);
-        if (errorEntry.getSourceRecordActionStatus() == JobLogEntry.ActionStatus.DISCARDED && errorEntry.getError().isEmpty()) {
-          errorEntry.setError(DATA_IMPORT_ERROR_DISCARDED);
-        }
-        if (!errorEntry.getError().isEmpty()) {
-          saveError(bulkOperationId, identifier, errorEntry.getError(), ErrorType.ERROR);
-        }
-      });
-    } catch (Exception e) {
-      log.error("Problem with retrieving logs from MetadataProvider", e);
-      throw new DataImportException(e);
+  public void saveErrorsFromDataImport(List<JobLogEntry> logEntries, BulkOperation bulkOperation) {
+    log.info("Saving errors from DataImport, total entries = {}", logEntries.size());
+    logEntries.stream()
+      .filter(entry -> nonNull(entry.getError()))
+      .forEach(entry -> processDataImportLogEntry(entry, bulkOperation));
+  }
+
+  private void processDataImportLogEntry(JobLogEntry errorEntry, BulkOperation bulkOperation) {
```

Comment:
I would add error to the name, `processDataImportErrorLogEntry`

Reply:
fixed